client/finality-grandpa/src/environment.rs

```diff
@@ -1178,6 +1182,19 @@ pub(crate) fn finalize_block<BE, Block, Client>(
 			},
 		};
 
+		// Q: We `finalized()` this at L37, so can I be sure
```

Comment:
Please address that by either:
- providing a proof in `expect`
- or printing an error if that's not the case.

TBH I'm more in favour of the second, since the notification is not super critical for node operation, so there is no point in panicking if the proof turns out to be invalid.

Reply:
Agreed. Let's just check for `Err` and log it under `debug`.